src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -68,7 +70,16 @@ public ModelManager(ReadOnlyAddressBook addressBook, UserPrefs userPrefs) {
         filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
         filteredWhitelistedPersons = new FilteredList<>(this.addressBook.getWhitelistedPersonList());
         filteredBlacklistedPersons = new FilteredList<>(this.addressBook.getBlacklistedPersonList());
+        filteredOverduePersons = new FilteredList<>(this.addressBook.getOverduePersonList());
+        for (ReadOnlyPerson person : allPersons) {
 
+            if (!person.getDeadline().value.equals("No deadline set.")) {
+                Date deadline = DateUtil.convertStringToDate(person.getDeadline().valueToDisplay);
+                if (deadline.before(new Date())) {
+                    this.addressBook.addOverdueDebtPerson(person);
+                }
+            }
+        }
```

Comment:
I dont think this method should be here law. As in, this is to check if there are any overdue persons and add them into the list. But by right, there shouldn't be right? In our sample data, we can make sure deadline is always in the future. What do you think about this?

Reply:
I'm putting it here for now. I will explain it to you offline. The error is very tedious to explain.